access/msteams/msapi/token.go

```diff
@@ -45,7 +45,7 @@ func (c *tokenWithTTL) Bearer(ctx context.Context, config Config) (string, error
 	expiresAt := c.expiresAt
 	c.mu.RUnlock()
 
-	if expiresAt == 0 || expiresAt > time.Now().UnixNano() {
+	if expiresAt == 0 || expiresAt < time.Now().UnixNano() {
```

Comment:
I think we should add an offset here
If the token expires in the next `retry count * timeout seconds` then renew the token.

As an alternative, we might change `access/msteams/msapi/client.go` to (possibly) renew the token when we receive a 401.

So, close to this `if` statement
```go
		if resp.StatusCode > 499 || resp.StatusCode == http.StatusTooManyRequests {
```

Reply:
:+1:

Reply:
Well, I was wrong and we renew the token 1 minute before expiry. This is conservative and should not cause us any time skew or slow call issue :)